pkg/commands/image/list_filter.go

```diff
@@ -0,0 +1,145 @@
+package image
+
+import (
+	corev1alpha1 "github.com/pivotal/kpack/pkg/apis/core/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"regexp"
+	"strings"
+
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+)
+
+const (
+	filterKindBuilder        = "builder"
+	filterKindClusterBuilder = "clusterbuilder"
+	filterKindLatestReason   = "latest-reason"
+	filterKindStatus         = "ready"
+)
+
+type filter struct {
```

Comment:
Could this be a interface with each filter type implementing it?

Reply:
I had considered it. Ultimately, I felt that an interface was a little too heavy because the functionality was only being used in one place. Moreover I thought the logic would feel more "spread out". I can be convinced otherwise.

Reply:
It strikes me as an easy way to avoid two switch statements.

Reply:
I'm personally still not won over. Interfaces add a layer of indirection which I generally find more painful than the switch statements as they are. I'm happy to be outvoted here.

Reply:
I'm a big fan of a little open closed principle to remove unnecessary switches if possible. Curious other people's thoughts. 

Another alternative could be to initialize the filter struct with its filter function
```
type filter struct {
	filterFunc func(image v1alpha1.Image, fs []filter) bool
	values []string
}
```

Reply:
Last thing I'll say on this. I'm not opposed to the open/closed principle just how eagerly I'd employ it. The [following article](http://joelabrahamsson.com/a-simple-example-of-the-openclosed-principle/#when-should-we-apply-the-openclosed-principle) articulates my thoughts a bit better:

> So, in other words, I definitely think we should have put some effort into abiding by the open/closed principle once the requirements started changing. Before that, in most cases, I would suggest limiting your efforts to ensuring that the code is well written enough so that it’s easy to refactor if the requirements starts changing.